test/conformance/main_test.go

```diff
@@ -38,10 +44,11 @@ func TestMain(m *testing.M) {
 			ComponentsToTest:    test.EventingFlags.Channels,
 		}
 		sourcesTestRunner = testlib.ComponentsTestRunner{
+			ComponentFeatureMap: testlib.SourceFeatureMap,
```

Comment:
Correct.

Reply:
`kn conformance run ...` ? Anyone?

Reply:
Would it be possible to restructure this so that something like this:
https://github.com/knative/eventing/pull/3531/files#diff-9cd17936c8cf7a4c529c64712ffa54abR37

Since these are conformance tests it would be nice if the user doesn't have to manually modify the test code to test their source for conformance. Ideally the creation of the resource would be separate from the tests and for tests, all you'd need to do is to give a namespace/name to test and the test would run against it. I've been advocating for this pattern for all of the conformance tests to make them easier to test. I further believe this to be even more important for sources as there are going to be more sources than channels / brokers.

Reply:
So it is my understanding that the "reusable" conformance tests package should not create any sources and should be "reusable" by other repos that want to run them against their sources, and that is the role of `conformance/helpers` package. Which is the case in this PR. 

The repo specific tests that reuse the helpers (in our case `test/conformance/main_tests.go & source_crd_metadata_test.go ` ) are the ones that inject the source creation tests. In `eventing-contrib` we'll do the same, inject the initializer that creates the `KafkaSource` etc and just reuse the conformance helper function. 

@vaikas is the above different from what you want?

Reply:
Well, Ideally I guess I'd like, just like today I can run E2E tests against my own cluster so I can push custom bits into it. I create my cluster, then ko apply my bits into it and then execute:
go test ./test/... -tags=e2e

I do not have to change any of the e2e tests to point to different cluster, nor have the tests have any knowledge how my bits got to the cluster.

Ideally what I think I'd like is the following experience for a user:
create their sources however they want (kubectl apply for example)
take the NS/Name of the above
then run something like:
go test ./test/... -tags=source-conformance -sourceNamespace=myns -sourceName=mysource
(or some shell script that executes that)

So, to run the tests I wouldn't have to modify the code at all. Maybe I'm missing why the helpers need to create these instead of having tests that take the namespace / name and execute the tests against it instead of creating the "units under test" in our repo being a requirement.

Reply:
As agreed I'll do the cmd args in a separate PR